modules/heat_conduction/src/materials/AnisoHeatConductionMaterial.C

```diff
@@ -10,137 +10,162 @@
-                                    isParamValid("thermal_conductivity_z_pp")
-                                ? &declareProperty<Real>("thermal_conductivity_z")
-                                : NULL),
-    _thermal_conductivity_z_dT(
-        _thermal_conductivity_z ? &declareProperty<Real>("thermal_conductivity_z_dT") : NULL),
-
-    _specific_heat(declareProperty<Real>("specific_heat")),
+template <bool is_ad>
+AnisoHeatConductionMaterialTempl<is_ad>::AnisoHeatConductionMaterialTempl(
+    const InputParameters & parameters)
+  : DerivativeMaterialInterface<Material>(parameters),
+    _ref_temp(getParam<Real>("reference_temperature")),
+
+    _has_temp(isCoupled("temperature")),
+    _T((_has_temp && !is_ad) ? coupledValue("temperature") : _zero),
+    _ad_T((_has_temp && is_ad) ? adCoupledValue("temperature") : _ad_zero),
+    _T_var(coupled("temperature")),
+    _T_name(getVar("temperature", 0)->name()),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+
+    _my_thermal_conductivity(getParam<std::vector<Real>>("thermal_conductivity")),
+    _my_specific_heat(isParamValid("specific_heat") ? getParam<Real>("specific_heat") : 500),
+
+    _thermal_conductivity(
+        declareGenericProperty<RankTwoTensor, is_ad>(_base_name + "thermal_conductivity")),
+    _dthermal_conductivity_dT(
+        declarePropertyDerivative<RankTwoTensor>(_base_name + "thermal_conductivity", _T_name)),
+    _thermal_conductivity_temperature_coefficient_function(
+        getParam<FunctionName>("thermal_conductivity_temperature_coefficient_function") != ""
+            ? &getFunction("thermal_conductivity_temperature_coefficient_function")
+            : nullptr),
+    _specific_heat(declareGenericProperty<Real, is_ad>(_base_name + "specific_heat")),
+    _dspecific_heat_dT(declarePropertyDerivative<Real>(_base_name + "specific_heat", _T_name)),
     _specific_heat_temperature_function(
         getParam<FunctionName>("specific_heat_temperature_function") != ""
             ? &getFunction("specific_heat_temperature_function")
-            : NULL)
+            : nullptr)
 {
-  bool k_x = isParamValid("thermal_conductivity_x") || (NULL != _thermal_conductivity_x_pp);
-  bool k_y = isParamValid("thermal_conductivity_y") || (NULL != _thermal_conductivity_y_pp);
-  bool k_z = isParamValid("thermal_conductivity_z") || (NULL != _thermal_conductivity_z_pp);
-
-  if (!k_x || (_subproblem.mesh().dimension() > 1 && !k_y) ||
-      (_subproblem.mesh().dimension() > 2 && !k_z))
+  if (!isParamValid("specific_heat") && !_specific_heat_temperature_function)
   {
-    mooseError("Incomplete set of orthotropic thermal conductivity parameters");
-  }
-  if (_specific_heat_temperature_function && !_has_temp)
-  {
-    mooseError("Must couple with temperature if using specific heat function");
+    mooseError("Either specific heat or specific heat temperature function must be defined");
```

Comment:
could you remove the `{ }` around single statement blocks please?

Reply:
Done